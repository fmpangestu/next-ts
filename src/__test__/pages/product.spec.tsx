// import ProductPage from "@/pages/product";
import ProductPage from "@/pages/product";
import { render } from "@testing-library/react";
jest.mock("next/router", () => {
  return {
    userRouter() {
      return {
        route: "/product",
        pathname: "",
        query: "",
        asPath: "",
        push: jest.fn(),
        events: {
          on: jest.fn(),
          off: jest.fn(),
        },
        beforePopState: jest.fn(),
        prefetch: jest.fn(() => null),
        isReady: true,
      };
    },
  };
});
describe("Product Page", () => {
  it("render product page", () => {
    const page = render(<ProductPage />);
    expect(page).toMatchSnapshot();
  });
});
