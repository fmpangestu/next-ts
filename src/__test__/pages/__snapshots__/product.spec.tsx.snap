// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Page render product page 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1
        class="text-3xl font-bold text-center my-5"
      >
        Product Page
      </h1>
      <div
        class="px-4 py-5 lg:py-16 lg:px-24 sm:grid sm:grid-cols-2 lg:grid-cols-4 gap-10 box-border"
      >
        <div
          class="w-full bg-gray-300 animate-pulse rounded-lg mb-4"
          data-testid="loading-placeholder"
        >
          <div
            class="h-48 bg-gray-900 rounded-t-lg"
          />
          <div
            class="px-3 py-2 w-full flex flex-col gap-2 mb-4 rounded-b-lg"
          >
            <div
              class="h-6 bg-gray-400 rounded"
            />
            <div
              class="h-4 bg-gray-400 rounded w-3/4"
            />
            <div
              class="h-4 bg-gray-400 rounded w-1/2"
            />
          </div>
        </div>
        <div
          class="w-full bg-gray-300 animate-pulse rounded-lg mb-4"
          data-testid="loading-placeholder"
        >
          <div
            class="h-48 bg-gray-900 rounded-t-lg"
          />
          <div
            class="px-3 py-2 w-full flex flex-col gap-2 mb-4 rounded-b-lg"
          >
            <div
              class="h-6 bg-gray-400 rounded"
            />
            <div
              class="h-4 bg-gray-400 rounded w-3/4"
            />
            <div
              class="h-4 bg-gray-400 rounded w-1/2"
            />
          </div>
        </div>
        <div
          class="w-full bg-gray-300 animate-pulse rounded-lg mb-4"
          data-testid="loading-placeholder"
        >
          <div
            class="h-48 bg-gray-900 rounded-t-lg"
          />
          <div
            class="px-3 py-2 w-full flex flex-col gap-2 mb-4 rounded-b-lg"
          >
            <div
              class="h-6 bg-gray-400 rounded"
            />
            <div
              class="h-4 bg-gray-400 rounded w-3/4"
            />
            <div
              class="h-4 bg-gray-400 rounded w-1/2"
            />
          </div>
        </div>
        <div
          class="w-full bg-gray-300 animate-pulse rounded-lg mb-4"
          data-testid="loading-placeholder"
        >
          <div
            class="h-48 bg-gray-900 rounded-t-lg"
          />
          <div
            class="px-3 py-2 w-full flex flex-col gap-2 mb-4 rounded-b-lg"
          >
            <div
              class="h-6 bg-gray-400 rounded"
            />
            <div
              class="h-4 bg-gray-400 rounded w-3/4"
            />
            <div
              class="h-4 bg-gray-400 rounded w-1/2"
            />
          </div>
        </div>
        <div
          class="w-full bg-gray-300 animate-pulse rounded-lg mb-4"
          data-testid="loading-placeholder"
        >
          <div
            class="h-48 bg-gray-900 rounded-t-lg"
          />
          <div
            class="px-3 py-2 w-full flex flex-col gap-2 mb-4 rounded-b-lg"
          >
            <div
              class="h-6 bg-gray-400 rounded"
            />
            <div
              class="h-4 bg-gray-400 rounded w-3/4"
            />
            <div
              class="h-4 bg-gray-400 rounded w-1/2"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <h1
      class="text-3xl font-bold text-center my-5"
    >
      Product Page
    </h1>
    <div
      class="px-4 py-5 lg:py-16 lg:px-24 sm:grid sm:grid-cols-2 lg:grid-cols-4 gap-10 box-border"
    >
      <div
        class="w-full bg-gray-300 animate-pulse rounded-lg mb-4"
        data-testid="loading-placeholder"
      >
        <div
          class="h-48 bg-gray-900 rounded-t-lg"
        />
        <div
          class="px-3 py-2 w-full flex flex-col gap-2 mb-4 rounded-b-lg"
        >
          <div
            class="h-6 bg-gray-400 rounded"
          />
          <div
            class="h-4 bg-gray-400 rounded w-3/4"
          />
          <div
            class="h-4 bg-gray-400 rounded w-1/2"
          />
        </div>
      </div>
      <div
        class="w-full bg-gray-300 animate-pulse rounded-lg mb-4"
        data-testid="loading-placeholder"
      >
        <div
          class="h-48 bg-gray-900 rounded-t-lg"
        />
        <div
          class="px-3 py-2 w-full flex flex-col gap-2 mb-4 rounded-b-lg"
        >
          <div
            class="h-6 bg-gray-400 rounded"
          />
          <div
            class="h-4 bg-gray-400 rounded w-3/4"
          />
          <div
            class="h-4 bg-gray-400 rounded w-1/2"
          />
        </div>
      </div>
      <div
        class="w-full bg-gray-300 animate-pulse rounded-lg mb-4"
        data-testid="loading-placeholder"
      >
        <div
          class="h-48 bg-gray-900 rounded-t-lg"
        />
        <div
          class="px-3 py-2 w-full flex flex-col gap-2 mb-4 rounded-b-lg"
        >
          <div
            class="h-6 bg-gray-400 rounded"
          />
          <div
            class="h-4 bg-gray-400 rounded w-3/4"
          />
          <div
            class="h-4 bg-gray-400 rounded w-1/2"
          />
        </div>
      </div>
      <div
        class="w-full bg-gray-300 animate-pulse rounded-lg mb-4"
        data-testid="loading-placeholder"
      >
        <div
          class="h-48 bg-gray-900 rounded-t-lg"
        />
        <div
          class="px-3 py-2 w-full flex flex-col gap-2 mb-4 rounded-b-lg"
        >
          <div
            class="h-6 bg-gray-400 rounded"
          />
          <div
            class="h-4 bg-gray-400 rounded w-3/4"
          />
          <div
            class="h-4 bg-gray-400 rounded w-1/2"
          />
        </div>
      </div>
      <div
        class="w-full bg-gray-300 animate-pulse rounded-lg mb-4"
        data-testid="loading-placeholder"
      >
        <div
          class="h-48 bg-gray-900 rounded-t-lg"
        />
        <div
          class="px-3 py-2 w-full flex flex-col gap-2 mb-4 rounded-b-lg"
        >
          <div
            class="h-6 bg-gray-400 rounded"
          />
          <div
            class="h-4 bg-gray-400 rounded w-3/4"
          />
          <div
            class="h-4 bg-gray-400 rounded w-1/2"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
